test/ntttcp_test.py

```diff
@@ -0,0 +1,470 @@
+        if buffer_size:
+            matched_buffer_size = buffer_size[0]
+            return int(matched_buffer_size)
+        else:
+            print("Can not match the buffer_size")
+
+    def is_output_log_file(self, filename) -> bool:
+        filename_out = open(filename, "r")
+        value_filename_out = filename_out.read()
+        throughput_in_logfile_pattern = re.compile(r"INFO:\s+throughput\s+:(\d+\.\d+)(\w+)")
+        throughput_in_logfile = throughput_in_logfile_pattern.findall(value_filename_out)
+        if subprocess.run(f"ls {filename}", shell=True, check=True) and throughput_in_logfile:
+            print("throughput_in_logfile:", throughput_in_logfile)
+            return True
+        else:
+            return False
+
+    def parseFile(self, filename) -> bool:
+        parser = make_parser()
+        parser.setContentHandler(ContentHandler())
+        parser.parse(filename)
+        print("parser.parse(filename):",parser.parse(filename))
+        if parser.parse(filename):
+            return True
+        else:
+            return False
+
+    def is_output_xml_file(self, filename) -> bool:
+        filename_out = open(filename, "r")
+        value_filename_out = filename_out.read()
+        throughput_in_xmlfile_pattern = re.compile(r"<throughput metric=\"MB\/s\">(\d+)\.(\d+)")
+        throughput_in_xmlfile = throughput_in_xmlfile_pattern.findall(value_filename_out)
+        if subprocess.run(f"ls {filename}", shell=True, check=True) and throughput_in_xmlfile:
+            print("throughput_in_xmlfile:", throughput_in_xmlfile)
+            try:
+                self.parseFile(filename)
+                return True
+            except Exception:
+                return False
+
+    def is_output_json_file(self, filename) -> bool:
+        filename_out = open(filename, "r")
+        value_filename_out = filename_out.read()
+        throughput_in_jsonfile_pattern = re.compile(r"\"throughputs\" : \[[\s\S]*\{[\s\S]*\"metric\" : \"MB\/s\"\,[\s\S]*\"value\" : \"(\d+\.\d+)\"")
+        throughput_in_jsonfile = throughput_in_jsonfile_pattern.findall(value_filename_out)
+        if subprocess.run(f"ls {filename}", shell=True, check=True) and json.loads(value_filename_out, strict=False):
+            if throughput_in_jsonfile:
+                print("throughput_in_jsonfile:", throughput_in_jsonfile)
+                return True
+            else:
+                return False
+        else:
+            return False
+
+    def get_starting_port_number(self) -> int:
+        # DBG : New connection: local:48536 [socket:4] --> 127.0.0.1:10000
+        starting_port_pattern = re.compile(r"DBG : New connection: local:(\d+) \[socket:4\] --> 127.0.0.1:(\d+)")
+        starting_port = starting_port_pattern.findall(self.send_stdout)
+        if starting_port:
+            print("starting_port:", starting_port[0][1])
+            return int(starting_port[0][1])
+        else:
+            return False
```

Comment:
The return value is int, not bool.
We can return None, if no matched starting_port. Then changed into Optional[int]

Reply:
done